cord19-explorer/src/main/java/edu/asu/diging/cord19/explorer/core/mongo/impl/PublicationSeachProviderImpl.java

```diff
@@ -0,0 +1,34 @@
+package edu.asu.diging.cord19.explorer.core.mongo.impl;
+
+import java.util.List;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.data.mongodb.core.MongoTemplate;
+import org.springframework.data.mongodb.core.query.Criteria;
+import org.springframework.data.mongodb.core.query.Query;
+import org.springframework.stereotype.Service;
+
+import edu.asu.diging.cord19.explorer.core.model.impl.PublicationImpl;
+import edu.asu.diging.cord19.explorer.core.mongo.PublicationSearchProvider;
+
+@Service
+public class PublicationSeachProviderImpl implements PublicationSearchProvider {
+
+    @Autowired
+    private MongoTemplate mongoTemplate;
+
+    @Override
+    public int searchResultSize(String title) {
+        Criteria regex = Criteria.where("metadata.title").regex(".*" + title + ".*", "i");
+        return mongoTemplate.find(new Query().addCriteria(regex), PublicationImpl.class).size();
```

Comment:
I think you want to use `count` here, not `find`. otherwise you still request all documents from the db, just get the count from the list (that's over 100,000 objects in production that you would load into memory).